doc/source/python.md

```diff
@@ -0,0 +1,496 @@
+managed delegate that will invoke the given Python callable when it
+is called:
+
+```python
+def my_handler(source, args):
+    print('my_handler called!')
+
+# instantiate a delegate
+d = AssemblyLoadEventHandler(my_handler)
+
+# use it as an event handler
+AppDomain.CurrentDomain.AssemblyLoad += d
+```
+
+Delegates with `out` or `ref` parameters can be implemented in Python by
+following the convention described in [Out and Ref parameters].
+
+Multicast delegates can be implemented by adding more callable objects to
+a delegate instance:
+
+```python
+d += self.method1
+d += self.method2
+d()
+```
+
+Events are treated as first-class objects in Python, and behave in many
+ways like methods. Python callbacks can be registered with event attributes,
+and an event can be called to fire the event.
+
+Note that events support a convenience spelling similar to that used in C#.
+You do not need to pass an explicitly instantiated delegate instance to an
+event (though you can if you want). Events support the `+=` and `-=`
+operators in a way very similar to the C# idiom:
+
+```python
+def handler(source, args):
+    print('my_handler called!')
+
+# register event handler
+object.SomeEvent += handler
+
+# unregister event handler
+object.SomeEvent -= handler
+
+# fire the event
+result = object.SomeEvent(...)
+```
+
+## Exception Handling
+
+You can raise and catch managed exceptions just the same as you would
+pure-Python exceptions:
+
+```python
+from System import NullReferenceException
+
+try:
+    raise NullReferenceException("aiieee!")
+except NullReferenceException as e:
+    print(e.Message)
+    print(e.Source)
+```
+
+## Using Arrays
+
+The type `System.Array` supports the subscript syntax in order to
+make it easy to create managed arrays from Python:
+
+```python
+from System import Array
+
+myarray = Array[int](10)
+```
+
+Managed arrays support the standard Python sequence protocols:
+
+```python
+items = SomeObject.GetArray()
+
+# Get first item
+v = items[0]
+items[0] = v
+
+# Get last item
+v = items[-1]
+items[-1] = v
+
+# Get length
+l = len(items)
+
+# Containment test
+test = v in items
+```
+
+Multidimensional arrays support indexing using the same notation one
+would use in C#:
+
+```python
+items[0, 2]
+
+items[0, 2, 3]
+```
+
+## Using Collections
+
+Managed arrays and managed objects that implement the IEnumerable
+interface can be iterated over using the standard iteration
+Python idioms:
+
+```python
+domain = System.AppDomain.CurrentDomain
+
+for item in domain.GetAssemblies():
+    name = item.GetName()
+```
+
+## Using COM Components
+
+Using Microsoft-provided tools such as `aximp.exe` and `tlbimp.exe`,
+it is possible to generate managed wrappers for COM libraries. After
+generating such a wrapper, you can use the libraries from Python just
+like any other managed code.
+
+Note: currently you need to put the generated wrappers in the GAC, in
+the PythonNet assembly directory or on the PYTHONPATH in order to load them.
```

Comment:
I think we should call this unsupported and direct people to one of Python packages that works with COM/WinRT directly.

Reply:
TBH, I would just remove that whole section ;)